File: source/code/csharp/account_management/cd_search.cs
var windowStart = new DateTimeOffset(2012, 01, 01, 00, 00, 00, TimeSpan.Zero);
var windowEnd = new DateTimeOffset(2015, 12, 31, 23, 59, 59, TimeSpan.Zero);

var accountsStartedBetween2012and2015 = await myDirectory
    .GetAccounts()
    .Where(a => (DateTimeOffset)a.CustomData["startDate"] >= windowStart
             && (DateTimeOffset)a.CustomData["startDate"] <= windowEnd)
    .Take(5)
    .ToListAsync();

File: source/code/vbnet/account_management/cd_search.vb
Dim windowStart = New DateTimeOffset(2012, 1, 1, 0, 0, 0, TimeSpan.Zero)
Dim windowEnd = New DateTimeOffset(2015, 12, 31, 23, 59, 59, TimeSpan.Zero)

Dim accountsStartedBetween2012and2015 = Await myDirectory _
    .GetAccounts() _
    .Where(Function(a) a.CustomData("startDate") >= windowStart _
               AndAlso a.CustomData("startDate") <= windowEnd) _
    .Take(5) _
    .ToListAsync()
